lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1233,11 +1222,34 @@ private synchronized void onConnected(ProtocolMessage message) {
             return;
         }
 
+        //RTN19a
+        sendPendingQueueMessages();
+
         /* indicated connected currentState */
         setSuspendTime();
         requestState(new StateIndication(ConnectionState.connected, error));
     }
 
+    /**
+     * Send all pending messages which are in the queue.
+     * Remove them from the queue once they are sent successfully
+     * Spec: RTN19a
+     */
+    private void sendPendingQueueMessages() {
```

Comment:
I think there are two issues here:
1) it needs to be possible, in the non-resume case, for messages to be reassigned a new `msgSerial` for the new connection

2) What happens to the messages where the `send()` failed with an exception? Are they left in the queue or should they be removed? And does the `CompletionListener` ever get called for them?

Reply:
Also - have you thought about adding some tests?

Reply:
> it needs to be possible, in the non-resume case, for messages to be reassigned a new msgSerial for the new connection

This should be happening https://github.com/ably/ably-java/pull/900/commits/62882a585949a428a9d6bf4f604f13fd4214df4c now

> What happens to the messages where the send() failed with an exception? Are they left in the queue or should they be removed? And does the CompletionListener ever get called for them?

I had not checked this as I used that portion of code directly from the other branch. But I think `QueuedMessage` itself should hold a reference to listener which seems to have been ignored here. 
I have passed the listener to send call 41ca4054be6b9a947cde6ffcbab8e2c35c07e6c9 but will check further whether SDK properly discards messages or keeps the messages in the queue. 

> Also - have you thought about adding some tests?
Yes, sorry I was going to but I manage to miss that. I will add any relevant tests